### ISSUE TITLE
Fails to create azure container right after storage account was created.

### ISSUE THREAD

**Issue by genevieve** [terraform-providers/terraform-provider-azurerm#841]
### Terraform Version
terraform v0.11.0
azure v1.1.1

### Affected Resource(s)
- azurerm_storage_container

### Terraform Configuration Files
```hcl
resource "azurerm_storage_account" "bosh" {
  name                = "${var.simple_env_id}"
  resource_group_name = "${azurerm_resource_group.bosh.name}"

  location                 = "${var.region}"
  account_tier             = "Standard"
  account_replication_type = "GRS"

  tags {
    environment = "${var.env_id}"
  }
}

resource "azurerm_storage_container" "bosh" {
  name                  = "bosh"
  resource_group_name   = "${azurerm_resource_group.bosh.name}"
  storage_account_name  = "${azurerm_storage_account.bosh.name}"
  container_access_type = "private"
}

resource "azurerm_storage_container" "stemcell" {
  name                  = "stemcell"
  resource_group_name   = "${azurerm_resource_group.bosh.name}"
  storage_account_name  = "${azurerm_storage_account.bosh.name}"
  container_access_type = "blob"
}

```

### Expected Behavior
It should have created the container.

This likely, like other azure resources, will require a retry here: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_storage_container.go#L108-L112

The azure-sdk only tries once: https://github.com/Azure/azure-sdk-for-go/blob/master/storage/container.go#L266-L275

### Actual Behavior
```
azurerm_storage_account.bosh: Still creating... (30s elapsed)
azurerm_storage_account.bosh: Creation complete after 36s (ID: /subscriptions/...-...oft.Storage/storageAccounts/bblciupenv)
azurerm_storage_container.bosh: Creating...
  container_access_type: "" => "private"
  name:                  "" => "bosh"
  properties.%:          "" => "<computed>"
  resource_group_name:   "" => "bbl-ci-up-env-bosh"
  storage_account_name:  "" => "bblciupenv"
azurerm_storage_container.stemcell: Creating...
  container_access_type: "" => "blob"
  name:                  "" => "stemcell"
  properties.%:          "" => "<computed>"
  resource_group_name:   "" => "bbl-ci-up-env-bosh"
  storage_account_name:  "" => "bblciupenv"
Error: Error applying plan:

2 error(s) occurred:

* azurerm_storage_container.bosh: 1 error(s) occurred:

* azurerm_storage_container.bosh: Error creating container "bosh" in storage account "bblciupenv": storage: service returned error: StatusCode=404, ErrorCode=ResourceNotFound, ErrorMessage=The specified resource does not exist.
...
* azurerm_storage_container.stemcell: 1 error(s) occurred:

* azurerm_storage_container.stemcell: Error creating container "stemcell" in storage account "bblciupenv": storage: service returned error: StatusCode=404, ErrorCode=ResourceNotFound, ErrorMessage=The specified resource does not exist.
...
```

### Steps to Reproduce
1. `terraform apply`

### References
There was a [fix](https://github.com/hashicorp/terraform/issues/7005) to ensure a storage container was not created until after the storage account was created and `Succeeded`. However, due to an eventually consistent api and a lack of retrying in the `storage_container` creation, this still fails.

**Comment by genevieve** [comment]
@tombuildsstuff Would you accept a PR for this functionality or do you want to see if there's anything that could be worked on in the sdk or api side for this issue?

**Comment by tombuildsstuff** [comment]
@genevieve so we'd definitely accept a PR for this, we should be retrying here.

It's worth noting that longer term [there's a new Azure Storage SDK available](https://github.com/Azure/azure-storage-blob-go) which we'll be migrating over too - but it's still early days. At this point we're holding off for stabilization/feature parity with the existing Storage SDK - my guess would be we'd migrate over to that sometime this summer.

Thanks!

**Comment by genevieve** [comment]
PR made!

**Comment by tombuildsstuff** [comment]
Closing since this was fixed via #846

**Comment by bgarcial** [comment]
According to some answer [here](https://stackoverflow.com/a/56697421/2773461) ...
>Creation of a storage account is an asynchronous process. When you execute az storage account create to create a storage account, request is sent to Azure and you get an accepted response back (if everything went well).

>The whole process to create (provision) a storage account takes some time (maximum 1 minute in my experience) and until the storage account is provisioned, no operations are allowed on that storage account.

How can I include a waiting process after the storage account creation? 
Particularly I am working with bash to deploy the environment, and It seems that only with the bash `sleep` command or pausing the bash shell with the`read` command for some time is not enough
